src/dft/descriptor.cxx

```diff
@@ -26,39 +27,74 @@ namespace detail {
 
 template <precision prec, domain dom>
 void descriptor<prec, dom>::set_value(config_param param, ...) {
-    int err = 0;
     va_list vl;
     va_start(vl, param);
     switch (param) {
-        case config_param::INPUT_STRIDES: [[fallthrough]];
+        case config_param::FORWARD_DOMAIN:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        case config_param::DIMENSION:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        case config_param::LENGTHS: {
+            if (values_.rank == 1) {
+                values_.dimensions = std::vector<std::int64_t>{ va_arg(vl, std::int64_t) };
+            }
+            else {
+                auto ptr = va_arg(vl, std::int64_t*);
+                std::copy(ptr, ptr + values_.rank + 1, values_.dimensions.begin());
```

Comment:
We should probably do a nullptr check here as well

Reply:
Done in [0b9d613](https://github.com/hjabird/oneMKL/pull/3/commits/0b9d61393167e2fcb17367e7a0decdf280c85f73).